tests/components/brother/test_sensor.py

```diff
@@ -0,0 +1,269 @@
+    assert state.attributes.get(ATTR_ICON) == "mdi:file-document-outline"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UNIT_PAGES
+    assert state.state == "538"
+
+    entry = registry.async_get("sensor.hl_l2340dw_duplex_unit_pages_counter")
+    assert entry
+    assert entry.unique_id == "0123456789_duplex_unit_pages_counter"
+
+    state = hass.states.get("sensor.hl_l2340dw_b_w_counter")
+    assert state
+    assert state.attributes.get(ATTR_ICON) == "mdi:file-document-outline"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UNIT_PAGES
+    assert state.state == "709"
+
+    entry = registry.async_get("sensor.hl_l2340dw_b_w_counter")
+    assert entry
+    assert entry.unique_id == "0123456789_b/w_counter"
+
+    state = hass.states.get("sensor.hl_l2340dw_color_counter")
+    assert state
+    assert state.attributes.get(ATTR_ICON) == "mdi:file-document-outline"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UNIT_PAGES
+    assert state.state == "902"
+
+    entry = registry.async_get("sensor.hl_l2340dw_color_counter")
+    assert entry
+    assert entry.unique_id == "0123456789_color_counter"
+
+    state = hass.states.get("sensor.hl_l2340dw_uptime")
+    assert state
+    assert state.attributes.get(ATTR_ICON) == "mdi:timer"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TIME_DAYS
+    assert state.state == "48"
+
+    entry = registry.async_get("sensor.hl_l2340dw_uptime")
+    assert entry
+    assert entry.unique_id == "0123456789_uptime"
+
+
+async def test_availability(hass):
+    """Ensure that we mark the entities unavailable correctly when device is offline."""
+    await init_integration(hass)
+
+    state = hass.states.get("sensor.hl_l2340dw_status")
+    assert state
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "waiting"
+
+    future = utcnow() + timedelta(minutes=5)
+    with patch("brother.Brother._get_data", side_effect=ConnectionError()):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("sensor.hl_l2340dw_status")
+        assert state
+        assert state.state == STATE_UNAVAILABLE
+
+    future = utcnow() + timedelta(minutes=10)
+    with patch(
+        "brother.Brother._get_data",
+        return_value=json.loads(load_fixture("brother_printer_data.json")),
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("sensor.hl_l2340dw_status")
+        assert state
+        assert state.state != STATE_UNAVAILABLE
+        assert state.state == "waiting"
+
+
+async def test_manual_update_entity(hass):
+    """Test manual update entity via service homeasasistant/update_entity."""
+    await init_integration(hass)
+
+    await async_setup_component(hass, "homeassistant", {})
+    with patch.object(
+        BrotherDataUpdateCoordinator, "_async_update_data", return_value=True
```

Comment:
Please patch the library instead. Don't patch our code. The coordinator is just a detail and may change in the future.

Reply:
Done.